src/memray/reporters/flamegraph.py

```diff
@@ -75,6 +105,21 @@ def __init__(
         self.data = data
         self.memory_records = memory_records
 
+    @classmethod
+    def update_import_system_frames_for_inverted_flamegraph(
+        cls, nodes: Sequence[FrameNodeDict], node_index: int
+    ) -> bool:
+        if not nodes or not nodes[node_index]:
+            return False
+
+        for child_index in nodes[node_index]["children"]:
+            result = cls.update_import_system_frames_for_inverted_flamegraph(
+                nodes, child_index
+            )
+            nodes[node_index]["import_system"] |= result
```

Comment:
The `|=` here is essentially saying: if any of this node's callers are part of the import system, then this is an import system node. That doesn't seem correct to me. Consider a test case like this:
```py
    def test_inverted_allocator_sometimes_called_under_importlib(self):
        # GIVEN
        peak_allocations = [
            MockAllocationRecord(
                tid=1,
                address=0x1000000,
                size=1024,
                allocator=AllocatorType.MALLOC,
                stack_id=1,
                n_allocations=1,
                _stack=[
                    ("malloc", "malloc.c", 12),
                    ("parent_one", "fun.py", 8),
                    ("main", "main.py", 4),
                ],
            ),
            MockAllocationRecord(
                tid=1,
                address=0x1000000,
                size=1024,
                allocator=AllocatorType.MALLOC,
                stack_id=1,
                n_allocations=1,
                _stack=[
                    ("malloc", "malloc.c", 12),
                    ("importer", "<frozen importlib>", 4),
                    ("parent_two", "fun.py", 10),
                    ("main", "main.py", 4),
                ],
            ),
        ]
```

In that example, `malloc` can be called from two different paths - either by `main` calling `parent_one` calling `malloc`, or by `main` calling `parent_two` calling `importer` calling `malloc`. `importer` is part of the import system, so there's one path to `malloc` that goes through the import system and one that doesn't. Each of those paths are responsible for half the bytes allocated by `malloc`.

Our non-inverted flame graph displays these stacks for that:
```text
|                root size=2048               |
|                main size=2048               |
| parent_one size=1024 | parent_two size=1024 |
|   malloc size=1024   |  importer size=1024  |
|                      |   malloc size=1024   |
```

When you click the "hide import system frames" checkbox, `importer` and the frames below it are hidden, and `parent_two` becomes a leaf:
```
|                root size=2048               |
|                main size=2048               |
| parent_one size=1024 | parent_two size=1024 |
|   malloc size=1024   |                      |
```

For our inverted flame graphs, I expect the exact same stacks to be shown, just with leaves and roots swapped. So, when import system frames are not hidden, I expect us to show this:
```text
|                root size=2048               |
|               malloc size=2048              |
| parent_one size=1024 |  importer size=1024  |
|    main size=1024    | parent_two size=1024 |
|                      |    main size=1024    |
```
But when import system frames are hidden, I expect it to instead show:
```text
|                root size=2048               |
|   malloc size=1024   | parent_two size=1024 |
| parent_one size=1024 |    main size=1024    |
|    main size=1024    |                      |
```

The `main -> parent_one -> malloc` path remains unchanged by "hide import system frames", since that path doesn't go through the import system, but the `main -> parent_two -> importer -> malloc` stack gets truncated to `main -> parent_two` when `importer` is reached (just like it is for non-inverted flame graphs), and the stack we show now ends with `parent_two` as a child of the root (because `parent_two` is a leaf in the non-inverted hide-import-system case, and inverted mode swaps leaves and roots). Which means that the `malloc` node shrinks from 2048 bytes to 1024 bytes, since half of its allocations are attributable to a path through the import system, and half aren't.

There's no way we can achieve that desired behavior if we update the `malloc` node to say that it's an import system frame because one of its callers is an import system frame, even though the other isn't. As this function is currently implemented, I think it would result in us hiding `malloc` entirely, effectively truncating the `main -> parent_one -> malloc` call stack even though it didn't pass through the import system.

I can think of 3 reasonable ways to handle this:
1. Build two separate trees in `--inverted` mode. For one, use full stacks as the input. For the other, truncate the stacks at import system calls before processing them into a tree. Include both trees in the inverted-mode HTML, and have the checkbox switch which one is rendered.
2. In the `FrameNodeDict`, instead of recording just a `bool` for whether or not the node was reached by the import system, record an `int`: the number of bytes allocated at that node through a path that went through the import system. That would change this `|=` into a `+=`. When the checkbox is unset the inverted-mode JS code would ignore that field. When the checkbox is checked the JS code would shrink every node by treating that node's size as `value - import_system_bytes`, instead of as `value` bytes. Any node where `value - import_system_bytes` is 0 would be removed, and its children would be hoisted up to the root.
3. Pre-process each stack to identify which frames are below a call into the import system. Include the is-import-system flag as part of the node key, so that we have two different `malloc` nodes in the dumped tree, each associated with 1024 bytes. When the checkbox is checked, our JS code drops nodes with is-import-system set, and hoists their children up a level. When the checkbox is unchecked, our JS code would need to traverse the tree and merge nodes together if they have the same parent and same location, but one is flagged is-import-system and the other isn't.

Of these options, I think 1 is probably the simplest, and 3 is probably the most complex.  2 might require the fewest changes to what's been implemented so far...

@pablogsal Please check what I've said here, and see if a) you agree with me about what the checkbox should do, and b) if my proposed solutions seem reasonable.